### PR TITLE
Bouncer

### PR DESCRIPTION
**Pull request by Ty-Omaha**
Adds the bouncer job
Original PR #6610
Credit to Kyet

10 hours as crew to unlock

- Tosses unruly patrons out of the bar. So the bartender can focus on drinks-making. Can also function as a bouncer for another department that wants him, such as Medical. Member of the service department. CQC works only in bar / kitchen

- Starts with: black jumpsuit, armored vest, sunglasses, flash telebaton,CQC **(chef)** one pair of zipties. Does NOT get sec radio access, or a mindshield implant. Does NOT get any kind of gun.

- Access: maint, weapons permit (so beepsky won't bust him for picking up any weapon an assailant drops).
image

image

:cl:Ty-Omaha, Kyet
add: Adds the Bouncer job
/ :cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
This was tried before: https://github.com/ParadiseSS13/Paradise/pull/6610 

Still has the same problems in my opinion; the chef and the bartender both are well equipped to deal with riff-raff---anything more should really be the purview of security---their job does encompass more than "hunt dem valids".

**Comment by Ty-Omaha** [comment]
>This was tried before: #6610

I know. That's why I linked and credited it at the very top of the PR.

>Also, thumbs upping your own PR:
Shameful.

I can't tell if you're being sarcastic or not

**Comment by Carthusia** [comment]
Honestly, at times I've found myself as the chef acting as a bouncer of sorts when the kitchen and bar merge together, and we have enough food on the buffet, waiters and everyone is working and things are going smoothly enough for me to hang out in the main bar zone for awhile, play some music, and then you get that one guy who starts a fight.

Out of everyone working there, the chef is the most equipped to deal with bad patrons due to their ability to just CQC chokehold them, and give them the bums rush right out the front door, or stuff them down disposals. No shotgun required, no blood on the floor, no one gets freaked out by a shotgun going off.

It's a nice little flavor job

**Comment by Ty-Omaha** [comment]
>Though their CQC should be identical to the chefs, limited to the kitchen, freezer, and bar, nowhere else as their job is basically to keep angry frothing masses away from the food preppers and the drink mixers.

thats what it is

**Comment by Carthusia** [comment]
> > Though their CQC should be identical to the chefs, limited to the kitchen, freezer, and bar, nowhere else as their job is basically to keep angry frothing masses away from the food preppers and the drink mixers.
> 
> thats what it is

You note "Can also function as a bouncer for another department that wants him, such as Medical"

In your OP, this kind of implies their CQC also can work within other departments, id suggest this little bit be removed, as it wouldn't make sense for them to be a bouncer for medical, bouncers are strictly leisure protection

**Comment by Ty-Omaha** [comment]
It says chef beside the CQC

**Comment by Carthusia** [comment]
> It says chef beside the CQC

Ahhh, it still seems odd to keep that bit and imply they can work guarding another department in general.
A bouncer in this capacity really should be keeping angry people out of the bar/kitchen if need be and not wander off to guard medical, or cargo.

Security can handle the rest of the station, but when it comes to drunken bar brawls or people being tools within the bar, the bouncer is the guy to handle it first to keep securitys time from being wasted on petty bar shenanigans.

besides that little niggle I have with the wording, I do like the idea and hope this passes.

**Comment by Ty-Omaha** [comment]
I updated the description

File: code/game/jobs/job/support.dm [hunk 1]
```diff
@@ -450,3 +450,36 @@
...
+	l_ear = /obj/item/radio/headset/headset_service
+	glasses = /obj/item/clothing/glasses/sunglasses
+	belt = /obj/item/melee/classic_baton/telescopic
+	backpack_contents = list(
+		/obj/item/flash = 1,
+		/obj/item/restraints/handcuffs/cable/zipties = 1
+)
+/datum/outfit/job/bouncer/post_equip(mob/living/carbon/human/H)
+	..()
+	var/datum/martial_art/cqc/under_siege/CQC = new(null)
```
Comment (farie82) [inline code/game/jobs/job/support.dm]: just `new` is enough here

**Comment by Carthusia** [comment]
How would their SOP look?

They would need an accompanying article within the service area, I assume similar to the bartentender, just slot it in right under theirs. 

I feel a draft should be written up to be pushed along side the new job, so that it has all its bases covered.

Stuff like: "Just a rough draft rewording of the bartenders SOP as an example

1. The Bouncer is not permitted to carry their issued weapons outside the bar. However, they may obtain permission from the Head of Security to do. Open carrying of said equipment beyond assigned zones without authorization is grounds for equipment confiscation. Exceptions are made during major emergencies, such as Nuclear Operatives or Blob Organisms;

2. The Bouncer is permitted to use their equipment on unruly bar patrons in order to throw them out if they are being disruptive. They are not, however, permitted to apply lethal, or near-lethal force.

3. The Bouncer is exempt from legal ramifications when dutifully removing unruly (ie, overtly hostile) patrons from the Bar, provided, of course, they followed Guideline 2.

4. The Bouncer has full permission to forcefully throw out anyone who climbs over the bar counter without permission, up to and including personnel who may have access to the side windoor. They are not, however, permitted to do so if the person in question uses the door, or is on an active investigation.

**Comment by shawnhind** [comment]
Just something to consider.  You state in the PR that the bouncer won't be issued any guns however, the bartender has a shotgun and beanbag slugs.  Since the bouncer is the one who is responsible for protecting the bar and not the bartender, I foresee the bartender often giving the shotgun to the bouncer.  What impact would this have?  Should this be prevented or should it be acceptable?

**Comment by Carthusia** [comment]
> Just something to consider. You state in the PR that the bouncer won't be issued any guns, however, the bartender has a shotgun and beanbag slugs. Since the bouncer is the one who is responsible for protecting the bar and not the bartender, I foresee the bartender often giving the shotgun to the bouncer. What impact would this have? Should this be prevented or should it be acceptable?

I mean, overall this is the case for custom job bouncers as well, nothing to stop it, and their SOP does not forbid it.

Maybe it the bartenders SOP was updated to include a bit about them being FULLY responsible for that shotgun and anything done with it, unless it was stolen, AND reported to security.

Still nothing to really stop it though, and all in all, with what the bouncer gets, a shotgun is a downgrade in all honesty, Chef level CQC is more than enough to do what they need to do.

**Comment by Kyep** [comment]
IMHO, this would perhaps be better off repurposed into a "Medical Orderly" and be exclusive to medical. I agree Bar doesn't really need them now that Bartender has shotgun, chef has CQC, etc.

**Comment by Carthusia** [comment]
I dono, while I do see people being annoying in medical often, it's not as common for that annoyance to break out into a straight up donnybrook where the floors get covered in blood and chairs start flying, compared to within the bar where this occurs on almost on a per round basis.

The Bartender has a shotty, though this isn't incredibly useful beyond the guy who jumps the counter and you get them at point blank or if you can get a straight shot at range, if your lucky and he doesn't move. Still gota toss him out, which means moving into fist range and abandoning your post.

The chef has Mild-CQC, but their job is kitchen duty, they can of course at times help with disturbances, but then that diverts them away from their primary job, and of course the MANY times as a chef ive been beaten to a pulp or outright murdered by a tider or angry crew who don't like how I serve food, a designated guy within that bar who coulda jumped in and helped, would have made a difference.

Perhaps two of them? One who is an Orderly who is assigned within medical, and the other who acts as a Bouncer within the bar/kitchen zones? Their prime job is maintaining the peace within the two zones with the most amount of trouble that's not so great that security needs to divert resources to handle.

**Comment by Tokorizo** [comment]
This should be a Karma role, at least 15 minimum.  10 hours is still newbie playtimes and CQC can be overkill in the wrong hands. Karma is earned through playing the game correctly, Hours are not and anyone can receive it, even grief players. It also adds another incentive to earn karma.

**Comment by Carthusia** [comment]
Thats a good point, before I got the hang of things with CQC, i kept disarming, thinking it would push them over like normal disarming, little did I know, i was putting that guy into soft crit and had to bring him into medical.

So a Karmalock would help mitigate any possible problems relating to fresh-ish players getting ahold of something they don't understand that can cause some real harm if used improperly.

Another karma reward wouldn't hurt either.

**Comment by Kyep** [comment]
Don't make this a karma job.
For the moment, it must remain non-karma so it can be testmerged and evaluated.

**Comment by Carthusia** [comment]
Alright, though whats your thought on Two of these, one as an Orderly for medical, white scrubs, white gloves with one for the bar as a Bouncer?

**Comment by nok-ko** [comment]
@Carthusia:
Medbay does not need more people stopping folks from getting treatment! 😅 
It's bad enough that the triple doors to medbay aren't on emergency access by default already. If someone breaking in and doing crimes is a problem, Sec can usually spare an officer to hang out and keep watch.

**Comment by Carthusia** [comment]
Kyet is the one who suggested making it a medical position, my counter is adding a position to both medical and the bar due to them being hotspots for more issues that on average don't really require security to break it up, but someone on site would be handy.

In most situations though, do you really want security to have to spare an officer when they are more or less undermanned in every scenario? 100 people, 5-10 officers if their lucky, thats 1 officer to every 10-15 people.

Couple people being a tool in medical or the bar, spare an officer on any other code than green and that's a hit they may not be able to afford and time wasted.

**Comment by Kyep** [comment]
For now:
- Make it non-karma
- Give it one slot
- See if it proves useful in medical and/or bar during testmerge/etc
- If it does then we can consider adding more slots or variants of it
- But get the basic version run in testing first

**Comment by Ty-Omaha** [comment]
>Make it non-karma
Give it one slot

It's already like that.

**Comment by Necaladun** [comment]
Willing to test this, but needs SoP and various rules of engagement before that's done. Having someone who is "sec-light" needs special stuff to deal with it.

I really, really, don't want this to be a validhunty role.

**Comment by Carthusia** [comment]
I wrote up an example SOP for them above, its more or less just a rewording of the bartenders SOP with a couple non-relevant bits removed, more could be added should anyone wish to take a crack at it.

**Comment by Quantum-M** [comment]
For the Medical Orderly, why not just add a checkpoint to medbay, and randomly have one security officer assigned to medbay by default, otherwise there will most likely be issues with a non-security medical orderly causing issues with validhunting, and potential issues when it comes to it's equipment (e.g. stun equipment) if said role is viable to be a antag.

**Comment by Fox-McCloud** [comment]
Update soon, or this will be closed.

**Pull request closed without merging**